src/libraries/System.ComponentModel.Composition/src/System/ComponentModel/Composition/Hosting/AssemblyCatalog.cs

```diff
@@ -543,6 +543,8 @@ private void ThrowIfDisposed()
         private string GetDisplayName() =>
             $"{GetType().Name} (Assembly=\"{Assembly.FullName}\")";   // NOLOC
 
+        [UnconditionalSuppressMessage("Single file", "IL3000: Avoid accessing Assembly file path when publishing as a single file",
+            Justification = "Setting a CodeBase is not single file dangerous")]
```

Comment:
In general I would avoid "dangerous"... we can use for example "single file compatible" or similar terminology.